controllers/operandconfig.go

```diff
@@ -260,17 +263,30 @@ func (r *CommonServiceReconciler) updateOperandConfig(newConfigs []interface{})
 	// Checking all the common service CRs to get the minimal size
 	opconServices, err = r.getMinimalSizes(opconServices, ruleSlice)
 	if err != nil {
-		return err
+		return true, err
+	}
+
+	// isEqual := reflect.DeepEqual(existingOpconServices, opconServices)
+	isEqual := true
+
+	for _, opService := range opconServices {
+		existingOpService := getItemByName(existingOpconServices.([]interface{}), opService.(map[string]interface{})["name"].(string))
+		for cr, spec := range opService.(map[string]interface{})["spec"].(map[string]interface{}) {
+			existingCr := existingOpService.(map[string]interface{})["spec"].(map[string]interface{})[cr].(map[string]interface{})
+			if isEqual = reflect.DeepEqual(spec, existingCr); !isEqual {
+				break
+			}
+		}
```

Comment:
@horis233 
After do a `DeepEqual` for each individual CR's spec, it could correctly handle the size switch between large and medium. Also it is working fine when customizing the sizes of individual CRs.

But once the small profile is involved, the `isEqual` will always be `false`.

Reply:
Interesting...